sql/core.go

```diff
@@ -198,8 +198,8 @@ type IndexableTable interface {
 
 // Inserter allow rows to be inserted in them.
 type Inserter interface {
-	// Insert the given row.
-	Insert(*Context, Row) error
+	// Insert the given row. Returns true if the row did not exist.
+	Insert(*Context, Row) (bool, error)
```

Comment:
I would leave the interface alone for the time being -- makes it much easier to accept this PR back to the upstream.

The semantics of Ignore are actually more general than just "ignore if this row already exists" -- it's really about ignoring any error in a batch of inserts:

https://dev.mysql.com/doc/refman/5.5/en/insert.html

So I think for a complete solution, we shouldn't change this interface, but we should define error types that integrators can return under different circumstances (nullness violation, duplicate keys, etc). This isn't just a simple bug fix, but an interface change. Let's start with a test case that doesn't work currently, and start a discussion with them about the right way to fix it.

Reply:
So as per the conversation held IRL, we'll just skip this altogether for now and come back to it at a later date.